config/src/main/java/org/springframework/security/config/annotation/web/configurers/ExpressionUrlAuthorizationConfigurer.java

```diff
@@ -188,6 +190,14 @@ final ExpressionBasedFilterInvocationSecurityMetadataSource createMetadataSource
 			if (trustResolver != null) {
 				defaultHandler.setTrustResolver(trustResolver);
 			}
+			ApplicationContext context = http.getSharedObject(ApplicationContext.class);
+			if(context != null) {
+				String[] roleHiearchyBeanNames = context.getBeanNamesForType(RoleHierarchy.class);
+				if(roleHiearchyBeanNames.length == 1) {
```

Comment:
What if more than 1 RoleHierarchy bean is in the context? Should we default to the first one? Or should we throw a configuration exception? What makes more sense?